src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/ConsoleAnalyze.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Cli
+{
+    public class ConsoleAnalyze : IAppCommand
+    {
+        private readonly IUpgradeContextFactory _contextFactory;
+        private readonly IUpgradeStateManager _stateManager;
+        private readonly ILogger<ConsoleAnalyze> _logger;
+        private readonly IEnumerable<IAnalyzeResultProvider> _providers;
+
+        public ConsoleAnalyze(
+            IEnumerable<IAnalyzeResultProvider> analysisProviders,
+            IUpgradeContextFactory contextFactory,
+            IUpgradeStateManager stateManager,
+            ILogger<ConsoleAnalyze> logger)
+        {
+            _providers = analysisProviders;
+            _contextFactory = contextFactory ?? throw new ArgumentNullException(nameof(contextFactory));
+            _stateManager = stateManager ?? throw new ArgumentNullException(nameof(stateManager));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task RunAsync(CancellationToken token)
+        {
+            _logger.LogWarning("The Analyze command feature is still under development, expect things to be not fully functional at the moment");
+            using var context = await _contextFactory.CreateContext(token);
+
+            await _stateManager.LoadStateAsync(context, token);
+            var analzyerContext = new AnalyzeContext(context);
+            foreach (var provider in _providers)
+            {
+                await provider.AnalyzeAsync(analzyerContext, token);
+            }
```

Comment:
AFAICT, this is to get some preliminary plumbing in. The end goal is to have it go through some data store for different formats

Reply:
What @twsouthwick said :) This is to get the initial plumbing in, as part of the next PR I will be addressing the storing it in data structure instead of printing in console.